src/hotspot/share/opto/loopopts.cpp

```diff
@@ -1503,8 +1513,7 @@ void PhaseIdealLoop::split_if_with_blocks_post(Node *n) {
             // to fold a StoreP and an AddP together (as part of an
             // address expression) and the AddP and StoreP have
             // different controls.
-            BarrierSetC2* bs = BarrierSet::barrier_set()->barrier_set_c2();
-            if (!x->is_Load() && !x->is_DecodeNarrowPtr() && !x->is_AddP() && !bs->is_gc_barrier_node(x)) {
+            if (!x->is_Load() && !x->is_DecodeNarrowPtr()) {
```

Comment:
why this change?

Reply:
This reverts Christian's fix for [JDK-8260420](https://bugs.openjdk.java.net/browse/JDK-8260420) which is no longer required.

Reply:
Ok.